api/v1alpha1/tls_types.go

```diff
@@ -15,6 +15,20 @@ type ClientTLSSettings struct {
 	// +optional
 	ClientValidation *ClientValidationContext `json:"clientValidation,omitempty"`
 	TLSSettings      `json:",inline"`
+
```

Comment:
Could we explain here what's the default strategy used by Envoy/EG? No TLS resumption, Stateless TLS resumption, or Stateful TLS resumption?

Reply:
Would we add an intermediate layer `SessionResumption` here to group all the session related parameters?

Reply:
Added a list of "typed" resumption settings and included docs explaining the default EG behavior.